elixir/apps/web/lib/web/components/core_components.ex

```diff
@@ -881,7 +881,8 @@ defmodule Web.CoreComponents do
   end
 
   def identity_has_email?(identity) do
-    not is_nil(provider_email(identity)) or identity.provider.adapter == :email
+    not is_nil(provider_email(identity)) or identity.provider.adapter == :email or
```

Comment:
Indeed I assumed the provider_identifier was an email because in the form for manually provisonning an oidc user, it is explicitly said it must be an email: https://github.com/firezone/firezone/blob/main/elixir/apps/web/lib/web/live/actors/components.ex#L105

However, I don't think there is any input validation.

Should I add a regex or close this?

Reply:
@Intuinewin Is the use case here that you would like a nice email to send to your OIDC users?

If that's the case, it seems like we could just enable the Send Welcome Email button for `provider_identifier` that have `@`. If it fails, they won't receive it, which is not worse than the behavior that exists now.

If the email was typo'd or sent to the wrong person, they still won't be able to log in, correct?

So the only downside here is the edge case an admin has entered an "email" that is not deliverable.

Reply:
Yes my use case is: after manually adding an oidc identity I want to send the user a welcome message (before their first login) !

> If the email was typo'd or sent to the wrong person, they still won't be able to log in, correct?

Yes, the welcome email doesn't contain any sensitive information except the name and slug of the account